src/include/serial.i

```diff
@@ -298,18 +298,27 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_PAGE *page,
 	 * progress. Serialization is also needed so only one thread does the
 	 * obsolete check at a time.
 	 */
-	if (upd->next != NULL &&
-	    __wt_txn_visible_all(session, page->modify->obsolete_check_txn)) {
-		F_CAS_ATOMIC(page, WT_PAGE_RECONCILIATION, ret);
-		/* If we can't lock it, don't scan, that's okay. */
-		if (ret != 0)
-			return (0);
-		obsolete = __wt_update_obsolete_check(session, page, upd->next);
-		F_CLR_ATOMIC(page, WT_PAGE_RECONCILIATION);
-		if (obsolete != NULL) {
-			page->modify->obsolete_check_txn = WT_TXN_NONE;
-			__wt_update_obsolete_free(session, page, obsolete);
+	if (upd->next == NULL)
+		return (0);
+
+	if (page->modify->obsolete_check_txn != WT_TXN_NONE) {
+		if (!__wt_txn_visible_all(session, page->modify->obsolete_check_txn)) {
```

Comment:
We're in a bind here because C (specifically C90) can't distinguish between integers and booleans.  `__wt_txn_visible_all` would return a boolean if it was part of our standard, so using `!` here is reasonable, I think.